lib/algolia/middleware/base_url.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Algolia.Middleware.BaseUrl do
+  @behaviour Tesla.Middleware
+
+  @impl Tesla.Middleware
+  def call(env, next, _opts) do
+    hint = env.opts[:subdomain_hint]
+    curr_retry = env.opts[:curr_retry]
+
+    host =
+      case {hint, curr_retry} do
```

Comment:
we don't know the url?

Reply:
I'm not sure I understand the question. The purpose of this middleware is similar to `Tesla.Middleware.BaseUrl`, which is to set the scheme/host part of the URL. But we don't use the Tesla one because the logic here for determining the host we use is dynamic.

Reply:
Yeah the dynamic part is what confused me. Ok now I see that this is Algolia's requirement to round robin requests through a pool of urls.